tests/php/src/Validation/ValidationURLProviderTest.php

```diff
@@ -0,0 +1,386 @@
+		AMP_Options_Manager::update_option( Option::ALL_TEMPLATES_SUPPORTED, false );
+		$this->assertTrue( $this->call_private_method( $this->validation_url_provider, 'is_template_supported', [ $author_conditional ] ) );
+		$this->assertFalse( $this->call_private_method( $this->validation_url_provider, 'is_template_supported', [ $search_conditional ] ) );
+
+		AMP_Options_Manager::update_option( Option::SUPPORTED_TEMPLATES, [ $search_conditional ] );
+		$this->assertTrue( $this->call_private_method( $this->validation_url_provider, 'is_template_supported', [ $search_conditional ] ) );
+		$this->assertFalse( $this->call_private_method( $this->validation_url_provider, 'is_template_supported', [ $author_conditional ] ) );
+	}
+
+	/**
+	 * Test get_posts_by_type.
+	 *
+	 * @covers ::get_posts_by_type()
+	 */
+	public function test_get_posts_by_type() {
+		$number_posts_each_post_type = 20;
+		$post_types                  = get_post_types( [ 'public' => true ], 'names' );
+
+		foreach ( $post_types as $post_type ) {
+			// Start the expected posts with the existing post(s).
+			$query          = new WP_Query(
+				[
+					'fields'    => 'ids',
+					'post_type' => $post_type,
+				]
+			);
+			$expected_posts = $query->posts;
+
+			for ( $i = 0; $i < $number_posts_each_post_type; $i++ ) {
+				array_unshift(
+					$expected_posts,
+					self::factory()->post->create(
+						[
+							'post_type' => $post_type,
+						]
+					)
+				);
+			}
+
+			$actual_posts = $this->call_private_method( $this->validation_url_provider, 'get_posts_by_type', [ $post_type ] );
+			$this->assertEquals( $expected_posts, array_values( $actual_posts ) );
+
+			// Test with the $offset and $number arguments.
+			$offset       = 0;
+			$actual_posts = $this->call_private_method( $this->validation_url_provider, 'get_posts_by_type', [ $post_type, $offset, $number_posts_each_post_type ] );
+			$this->assertEquals( array_slice( $expected_posts, $offset, $number_posts_each_post_type ), $actual_posts );
+		}
+	}
+
+	/**
+	 * Test get_taxonomy_links.
+	 *
+	 * @covers ::get_taxonomy_links()
+	 */
+	public function test_get_taxonomy_links() {
+		$number_links_each_taxonomy = 20;
+		$taxonomies                 = get_taxonomies(
+			[
+				'public' => true,
+			]
+		);
+
+		foreach ( $taxonomies as $taxonomy ) {
+			// Begin the expected links with the term links that already exist.
+			$expected_links             = array_map( 'get_term_link', get_terms( [ 'taxonomy' => $taxonomy ] ) );
+			$terms_for_current_taxonomy = [];
+			for ( $i = 0; $i < $number_links_each_taxonomy; $i++ ) {
+				$terms_for_current_taxonomy[] = self::factory()->term->create(
+					[
+						'taxonomy' => $taxonomy,
+					]
+				);
+			}
+
+			// Terms need to be associated with a post in order to be returned in get_terms().
+			wp_set_post_terms(
+				self::factory()->post->create(),
+				$terms_for_current_taxonomy,
+				$taxonomy
+			);
```

Comment:
Maybe assert this call worked as intended?
```suggestion
			$result = wp_set_post_terms(
 				self::factory()->post->create(),
 				$terms_for_current_taxonomy,
 				$taxonomy
 			);
 			$this->assertFalse( is_wp_error( $result ) );
```